### ISSUE TITLE
[no-extra-non-null-assertion] incorrect report for assertion in an optional chain

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#2166]
**Repro**

cc @sindresorhus @fregante
https://github.com/sindresorhus/refined-github/pull/3168#discussion_r434478709

```JSON
{
  "rules": {
    "@typescript-eslint/no-extra-non-null-assertion": ["error"]
  }
}
```

```TS
checksCounter?.textContent!.trim()
```

**Expected Result**

No errors

**Actual Result**

```TS
   checksCounter?.textContent!.trim()
// ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Forbidden extra non-null assertion.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `master` |
| `@typescript-eslint/parser`        | `master` |
| `TypeScript`                       | `3.9.x` |
